Major refactor of tutorial folder structure.

morrieinmaas opened the pull request:
* In accordance with isseue 61 description on GitHub

TODOs:

* Fix all broken links within the notebooks
* extend notebooks to cover mediation and multitenant
* possibly introduce helper functions that cn be imported from extra
folder
* !!! If medation and multitenant live in folder 1. the
docker-compose.yaml and manage script need to be adjusted to accomodate
for that setup. TBD what that looks like. More info
[here](https://github.com/hyperledger/aries-cloudagent-python/blob/main/Multitenancy.md#multi-tenant-admin-api)

## Description
see [issue 61](https://github.com/OpenMined/PyDentity/issues/61)

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

morrieinmaas committed:
Major refactor of tutorial folder structure.

* In accordance with isseue 61 description on GitHub

TODOs:

* Fix all broken links within the notebooks
* extend notebooks to cover mediation and multitenant
* possibly introduce helper functions that cn be imported from extra
folder
* !!! If medation and multitenant live in folder 1. the
docker-compose.yaml and manage script need to be adjusted to accomodate
for that setup. TBD what that looks like. More info
[here](https://github.com/hyperledger/aries-cloudagent-python/blob/main/Multitenancy.md#multi-tenant-admin-api)

morrieinmaas committed:
Restructure tutorial notebooks.

* Created folder structure and numbering according to [issue
61](https://github.com/OpenMined/PyDentity/issues/61)
* Adjusted notebook crossreferences/links accordingly in all notebooks
* Fixed some broken links to respective other notebooks in notebooks

Fixed all links for 1. Learning Aries, ACA-py and the
Basic Controller for

* Alice's Notebooks
* Checked all links manually

Fixed all links for bob in 1 Basic concepts

* All manually checked

morrieinmaas committed:
Renamed folder and fixed all links for 2. Public Ledgers and Mobile
Wallets

morrieinmaas committed:
Remove multitenant args file

* This will be handled by parsing startup args like described [here](https://github.com/hyperledger/aries-cloudagent-python/blob/b8f28dfe1b8b84f1fd5143bd5219c93b0913c3cc/aries_cloudagent/config/argparse.py)

morrieinmaas committed:
WIP adding multitenancy tutorial

Basics to version-bump aca-py

* Bumped aca-py version to 0.6 in Dockerfile
* add auto-provision param to aries basic args
* add auto-provision param to aries advanced args

Adjust proof controller to reflect version 0.6 API of aca-py.

* Added sending problem report function
* Tidied up th ecode to reflect Python linting
* Tidied up function definitions in order to reflect swagger docs order

Bump version of aca-py for attachment protocol example

* Change aca-py version for Docker image
* add --auto-provision flag to aca-py start so it runs properly and a
new wallet is created on startup

Changes to plugin routes to work with v0.6.0

* Connection takes now a session instead of context
* session is now a context property (context.session())
* context now comes from AdminRequestContext (new import) and is the
request["context"]
* Outbound_handler now directly takes the router from request and not
app.request
* ConnectionRecord has changed to ConnRecord (this takes the session)

Add mediation controller

* Added all async functions to cover mediation via API
* minor fixes to proofs.py, fixed naming of parsed args

Removed superfluous variable parsing in mediation.py

Make changes to remove by topic method call without the dirty changes to
the notebook like cell runs

Add reference to mediation manual/README for usage

Remove duplicate auto-provion flag from docker-compose.yml

WIP Basic startup with multitenancy enabled

* Pass args in dockercompose yaml
* bump version of aca-py for ngrok
bump aiohttp version to 3.7.2

WIP Basic notebook setup

* create an basewallet agent
* create an external agent
* create notebook structure for each of the above
** Bob and Alice have separate folders because they each hold their own subwallets and can be considered different users on the same agent
* Renaming varibales in manage and docker-compose.yml accordingly in order to reflect basewallet and ecternal agents

morrieinmaas committed:
WIP Basic notebook setup

* create an basewallet agent
* create an external agent
* create notebook structure for each of the above
** Bob and Alice have separate folders because they each hold their own subwallets and can be considered different users on the same agent
* Renaming varibales in manage and docker-compose.yml accordingly in order to reflect basewallet and ecternal agents

Basics to version-bump aca-py

* Bumped aca-py version to 0.6 in Dockerfile
* add auto-provision param to aries basic args
* add auto-provision param to aries advanced args

Adjust proof controller to reflect version 0.6 API of aca-py.

* Added sending problem report function
* Tidied up th ecode to reflect Python linting
* Tidied up function definitions in order to reflect swagger docs order

Bump version of aca-py for attachment protocol example

* Change aca-py version for Docker image
* add --auto-provision flag to aca-py start so it runs properly and a
new wallet is created on startup

Changes to plugin routes to work with v0.6.0

* Connection takes now a session instead of context
* session is now a context property (context.session())
* context now comes from AdminRequestContext (new import) and is the
request["context"]
* Outbound_handler now directly takes the router from request and not
app.request
* ConnectionRecord has changed to ConnRecord (this takes the session)

Add mediation controller

* Added all async functions to cover mediation via API
* minor fixes to proofs.py, fixed naming of parsed args

Removed superfluous variable parsing in mediation.py

Make changes to remove by topic method call without the dirty changes to
the notebook like cell runs

Add reference to mediation manual/README for usage

morrieinmaas committed:
Merge branch 'master' into issue_61

wip-abramson commented:
Sorry to be a pain but wondering about a slight refactor of tutorial structure for the first tutorial.

Basically Proofs tutorials wont work unless they have first issued themselves a credential. So perhaps it makes sense just to condense into the Credentials folder?

I also think Out of Band tutorial is an Advanced concept not a messaging one.

Like the tutorial names though!

morrieinmaas committed:
Add basic multitenant functionality and notebook examples

* Extend the AriesController class to handle multitenancy
* Update the API for multiitenancy to work properly
** Made request body optional where possible by providing a default
* Create notebooks for Alice and Bob in order to
** Create two wallets on the same agent
** Cover all API endpoints for multi tenancy

Signed-off-by: morrieinmaas <motechnix@gmail.com>

morrieinmaas commented:
@wip-abramson I'll refactor like that. Makes sense to me.

morrieinmaas committed:
Address MR feedback to restructure tutorials folders

* Proofs tutorials now lives in Credentials folder
* Out of band tutorial is now in Advanced Concepts folder

morrieinmaas committed:
WIP trying to establish mediation with Alice and Bob on the same agent

lohanspies committed:
fix jwt_token

lohanspies committed:
clean up jtw debug prints

lohanspies committed:
minor fixes in mediation.py

lohanspies committed:
delete stale notebook

lohanspies committed:
rename and updated notebooks

lohanspies committed:
added reuse address and port

wip-abramson commented on configuration/aries-args-advanced.yaml:
```diff
@@ -57,3 +57,4 @@ debug-connections: true
 debug-credentials: true
 debug-presentations: true
 enable-undelivered-queue: true
+open-mediation: true
```
We should specify this in the docker-compose instead

wip-abramson commented on libs/aries-basic-controller/aries_basic_controller/aries_controller.py:
```diff
@@ -35,10 +37,13 @@ def __init__(
         action_menu: bool = True,
         revocations: bool = True,
         api_key: str = None,
+        jwt_token: str = None,
```
can we call this subwallet_token or something like that

wip-abramson commented on libs/aries-basic-controller/aries_basic_controller/aries_controller.py:
```diff
@@ -77,12 +94,23 @@ def __init__(
             )
 
+        
+    # def update_jwt_token(self, api_key: str):
```
I think we should have this.
something like set_subwallet_token

morrieinmaas committed:
WIP updating multitenancy notebooks

* Improve tutorial structure (WIP)
* Extend AriesAgentController class to have a method that updates the
JWT token

morrieinmaas committed:
WIP Wee update in Bob Tutorial NB. Websocket showcase

* Little more structure to Bob's tutorial Notebook. Still needs work
* Check out the examples in the Websocket notebooks in the Bob folder
** This is a minimal-ish implementation of a Websocket server instead of
http that can handle multiple client connections to solve the current
webhook port-in-use problem.
** It already demonstrates running multiple clients on the same
endpoint. Here 127.0.0.1:8090/ws.
** I reckon we can make this quite elegant by re-writing this further
and using routes like /ws/{some-unique-wallet-id}/the-webhook-handle to
send specific webhooks only to the actual recepient.
** For the previous point could even use authentication with that (not
sure that's required actually because the agent should already be only
able to connect when having the right credential?!) TODO: investigate
this point.

wip-abramson committed:
add mediator agent

wip-abramson committed:
update tutorial

morrieinmaas committed:
Fixing bug in aries controller ClientSession so it does not get
instantiated twice.

* Rewrite header assignment logic
* Minor changes in notebooks

morrieinmaas committed:
Properly working tutorials Part 1 and 2 for mediation

* Fix mediation API request_mediation to not require passing an empty
object
* Tidy up notebooks part 1 and 2
* Checked functionality of notebooks
* Remove superfluous notebooks

TODO: Finish part 3 of the tutorial notebooks

morrieinmaas committed:
WIP Almost finished with notebooks

* Still need to accept credential

morrieinmaas committed:
Multitenant tutorial working

* Finished up notebooks
* Changed startup logic to accomodate multitenent use and mediation
** This means passing a different env yaml file by setting a flag in
manage

morrieinmaas committed:
Clean up notebooks

morrieinmaas committed:
Merge branch 'master' into issue_61

morrieinmaas committed:
Minor fixes and tidying

morrieinmaas committed:
Working complete multiteant tutorials

* manually checked

morrieinmaas committed:
Remove redundant dependency

morrieinmaas committed:
Tidying up and fixing logical error in token assignment in aries
controller

morrieinmaas commented:
@wip-abramson @lohanspies Please have a look. I'm happy for any feedback. Lesson learned, pulled too much into one task. that got a bit out of hand volume-wise.

morrieinmaas committed:
Remove unwanted tutorial notebooks of om pki lesson

lohanspies commented:
> at got a bit out of hand volume-wise.

Agree. Let us keep it focussed on small achievable chunks.

wip-abramson committed:
clean multi-tennant tutorial

wip-abramson commented on configuration/aries-args-multitenant.yaml:
```diff
@@ -0,0 +1,60 @@
+debug-credentials: true
+debug-presentations: true
+enable-undelivered-queue: true
+open-mediation: true
```
We should move this out of here

wip-abramson reviewed (approved):
All looks good to me, good work
